### ISSUE TITLE
Asking about inference result

### ISSUE THREAD

**Issue by YuyangXueEd** [KHU-AGI/GUIDE#2]
Hello, thank you very much for your great work and congratulations on the CVPR acceptance.

When reimplementing your code, I found there is no inference code for the unlearned model, so I adapted from `unlearn.py`. I used both the provided aligned face from `data/CelebAHQ/512` and some out of distribution face to do the unlearning and use the `last.pkl` model to do inference on FFHQ in-distribution face examples. I found that all face that do not belong to the unlearned identity also changed, which is contradict to Figure 8. in the paper. Could you please enlight me why is this happen?

Here is one example:

![image](https://github.com/user-attachments/assets/287eae03-a45d-493d-a1fc-4c2ea6201291)

Not sure why all these faces are changed when using the unlearned model. I used `loss + adj + global` loss to train the model for 1000 steps.

Or could you please share the standard code to inference, with the pretrained unlearned model?

Looking forward to your reply. Thank you very much!

**Comment by YuyangXueEd** [comment]
And also may I ask that I use two different image to unlearn the model, however the generated unlearned face is 99% similar through some similarity detection model. However, the two images' direction vector should be different (though I used both d=30). Any idea why this happened?

![image](https://github.com/user-attachments/assets/7c5f80a8-dcb0-4a75-9672-0bd56a9c33a3)

Best.

**Comment by JJuOn** [comment]
For the first comment, you can sample randomly generated images using `evaluate_fid.py`.
This code includes sampling from both pre-trained and unlearned models (EG3D and GUIDE) and computing FID scores.
The results are related to Figure 8 in our main paper.

Regarding the attached figures, you have tried to infer other identities on the unlearned model.
The model seems to be conducted unlearning on an image at the first row, and second column.
I am wondering if you processed alignment for the images at the first, third and fourth columns.
All of the images we used in our paper are processed using [Deep3DFaceRecon](https://github.com/sicxu/Deep3DFaceRecon_pytorch).
If you use aligned image, you can get better results.

For the second comment, we agree that the unlearned samples from the different source identities look similar.
However, how similar the unlearned images are not our main concern.
Instead, in our work, we focused on how to make the source and unlearned images dissimilar.

Nevertheless, future work on your point would be interesting.

**Comment by YuyangXueEd** [comment]
> For the first comment, you can sample randomly generated images using `evaluate_fid.py`. This code includes sampling from both pre-trained and unlearned models (EG3D and GUIDE) and computing FID scores. The results are related to Figure 8 in our main paper.
> 
> Regarding the attached figures, you have tried to infer other identities on the unlearned model. The model seems to be conducted unlearning on an image at the first row, and second column. I am wondering if you processed alignment for the images at the first, third and fourth columns. All of the images we used in our paper are processed using [Deep3DFaceRecon](https://github.com/sicxu/Deep3DFaceRecon_pytorch). If you use aligned image, you can get better results.
> 
> For the second comment, we agree that the unlearned samples from the different source identities look similar. However, how similar the unlearned images are not our main concern. Instead, in our work, we focused on how to make the source and unlearned images dissimilar.
> 
> Nevertheless, future work on your point would be interesting.

Hello, thank you very much for your prompt reply.

I conducted some further experiments, because I want to replicate Figure 8. from your paper. However, I cannot reproduce them as the face are largely changed. To ensure the preservation, I changed `global_loss_labmda` to `10`. However, still the face is changed. I wonder whether the inversion model cannot faithfully restore the face in the latent space.

![image](https://github.com/user-attachments/assets/f0419514-4a47-4658-927f-0d8aaaf70e49)

Is there any possible that you provide how to reproduce Figure 8.? It puzzles me a lot.

**Comment by TY-LEE-KR** [comment]
Thank you for your interest in our work.

It seems that your experiment involved inversion using GOAE. The results of Table 8 were conducted with the same random noise (and using mapping network). This was done to evaluate the generation performance of the EG3D model itself. It appears that the setup of your experiment differs from ours.

We believe reviewing this aspect should help clarify any questions you may have.
